pkg/handler/v2/application_installation/application_installation_test.go

```diff
@@ -0,0 +1,480 @@
+							Name: "sample-app",
+							Version: appskubermaticv1.Version{
+								Version: *semverlib.MustParse("v1.0.0"),
+							},
+						},
+					},
+				},
+				{
+					ObjectMeta: apiv1.ObjectMeta{
+						Name: "app2",
+					},
+					Namespace: app2TargetNamespace,
+					Spec: &appskubermaticv1.ApplicationInstallationSpec{
+						Namespace: appskubermaticv1.NamespaceSpec{
+							Name:   app2TargetNamespace,
+							Create: true,
+						},
+						ApplicationRef: appskubermaticv1.ApplicationRef{
+							Name: "sample-app",
+							Version: appskubermaticv1.Version{
+								Version: *semverlib.MustParse("v1.0.0"),
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.Name, func(t *testing.T) {
+			requestURL := fmt.Sprintf("/api/v2/projects/%s/clusters/%s/applicationinstallations", tc.ProjectID, tc.ClusterID)
+			req := httptest.NewRequest(http.MethodGet, requestURL, strings.NewReader(""))
+			res := httptest.NewRecorder()
+
+			ep, err := test.CreateTestEndpoint(*tc.ExistingAPIUser, nil, tc.ExistingKubermaticObjects, nil, hack.NewTestRouting)
+			if err != nil {
+				t.Fatalf("failed to create test endpoint due to: %v", err)
+			}
+
+			ep.ServeHTTP(res, req)
+
+			if res.Code != tc.ExpectedHTTPStatusCode {
+				t.Errorf("Expected HTTP status code %d, got %d: %s", tc.ExpectedHTTPStatusCode, res.Code, res.Body.String())
+				return
+			}
+
+			actualApplicationInstallations := test.NewApplicationInstallationWrapper{}
+			actualApplicationInstallations.DecodeOrDie(res.Body, t).Sort()
+
+			wrappedExpectedApplicationInstallations := test.NewApplicationInstallationWrapper(tc.ExpectedResponse)
+			wrappedExpectedApplicationInstallations.Sort()
+
+			actualApplicationInstallations.EqualOrDie(wrappedExpectedApplicationInstallations, t)
+		})
+	}
+}
+
+// TODO I am undecided if we need any more testcases here. They all seem to be standard cases, which are
+// handled by generic errors (e.g. object already exists, invalid spec).
+func TestCreateApplicationInstallation(t *testing.T) {
+	t.Parallel()
+	testcases := []struct {
+		Name                      string
+		ProjectID                 string
+		ClusterID                 string
+		ExistingKubermaticObjects []ctrlruntimeclient.Object
+		ExistingAPIUser           *apiv1.User
+		ApplicationInstallation   *apiv2.ApplicationInstallation
+		ExpectedResponse          *apiv2.ApplicationInstallation
+		ExpectedHTTPStatusCode    int
+	}{
+		{
+			Name:      "create ApplicationInstallation that matches spec",
+			ProjectID: test.GenDefaultProject().Name,
+			ClusterID: test.GenDefaultCluster().Name,
+			ExistingKubermaticObjects: test.GenDefaultKubermaticObjects(
+				test.GenTestSeed(),
+				test.GenDefaultCluster(),
+			),
+			ExistingAPIUser:         test.GenDefaultAPIUser(),
+			ApplicationInstallation: test.GenApiApplicationInstallation("app1", test.GenDefaultCluster().Name, app1TargetNamespace),
+			ExpectedHTTPStatusCode:  http.StatusOK,
```

Comment:
I will pass once you apply my suggestion from `routes_v2.go`
```suggestion
			ExpectedHTTPStatusCode:  http.StatusCreated,
```